code/modules/mob/living/living.dm

```diff
@@ -884,6 +884,20 @@
 
 	apply_effect((amount*RAD_MOB_COEFFICIENT)/max(1, (radiation**2)*RAD_OVERDOSE_REDUCTION), EFFECT_IRRADIATE, blocked)
 
+/mob/living/anti_magic_check(magic = TRUE, holy = FALSE)
+	if(!magic && !holy)
+		return
+
+	if((magic && has_trait(TRAIT_ANTIMAGIC)) || (holy && has_trait(TRAIT_HOLY)))
+		return src
+
+	var/list/obj/item/item_list = list()
+	item_list |= get_equipped_items(TRUE)
+	item_list |= held_items
```

Comment:
can be simplified as `var/list/obj/item/item_list = held_items | get_equipped_items(TRUE)`